plutus-tx-plugin/src/PlutusTx/Compiler/Expr.hs

```diff
@@ -405,7 +405,16 @@ hoistExpr var t =
                     (PIR.Def var' (PIR.mkVar () var', PIR.Strict))
                     mempty
 
-                t' <- compileExpr t
+                CompileContext {ccOpts=profileOpts} <- ask
+                t' <- do
+                    if coProfile profileOpts==All then do
+                        let ty = PLC._varDeclType var'
+                            varName = T.pack $ show $ PLC._varDeclName var'
+                        t'' <- compileExpr t
+                        lamName <- PLC.freshName "lamName"
```

Comment:
I often call these unit arguments "thunk" or "dead" or something. I kind of want to make `_` a valid name so we can use that!